### DISCUSSION TITLE
MacPorts update issue

### DISCUSSION THREAD

**Discussion by ihhub** [ihhub/fheroes2#5404] [General]
One user reports that they cannot update fheroes2 via macPorts:
```
I have tried "sudo port upgrade fheroes2", "sudo port install fheroes2" and "sudo port selfupdate && sudo port upgrade fheroes2".
It just keeps saying...

--->  Computing dependencies for fheroes2
--->  Cleaning fheroes2
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.
```

Hi @oleg-derevenetz , could you please help with solving this issue with MacPorts? Maybe something trivial is missed.

**Comment by oleg-derevenetz** [comment]
He should update the ports tree first using the `port selfupdate` command. 

P.S. To be honest, I don't understand exactly what the problem is. According to the output above, `fheroes2` has already been updated to the latest version, so there is simply nothing to do for the `port upgrade` command.

**Comment by JamonBull** [comment]
Hi @oleg-derevenetz. I have done the "port selfupdate" command. "sudo port selfupdate && sudo port upgrade fheroes2" still doesn't do anything. Terminal still launches 0.9.14. Not sure what to do next. Thanks.

**Comment by oleg-derevenetz** [comment]
Hi @JamonBull

Hmm, works for me:

```
oleg@MacBook-Pro ~ % fheroes2
Free Heroes of Might and Magic II, version: 0.9.14
                                                                                                                                                                                                            
oleg@MacBook-Pro ~ % sudo port selfupdate && sudo port upgrade fheroes2
oleg@MacBook-Pro ~ % sudo port selfupdate && sudo port upgrade fheroes2
--->  Updating MacPorts base sources using rsync
MacPorts base version 2.7.2 installed,
MacPorts base version 2.7.2 downloaded.
--->  Updating the ports tree
--->  MacPorts base is already the latest version

The ports tree has been updated. To upgrade your installed ports, you should run
  port upgrade outdated
--->  Fetching archive for curl-ca-bundle
--->  Attempting to fetch curl-ca-bundle-7.83.1_0.darwin_21.noarch.tbz2 from https://packages.macports.org/curl-ca-bundle
--->  Attempting to fetch curl-ca-bundle-7.83.1_0.darwin_21.noarch.tbz2.rmd160 from https://packages.macports.org/curl-ca-bundle
--->  Installing curl-ca-bundle @7.83.1_0
--->  Cleaning curl-ca-bundle
--->  Deactivating curl-ca-bundle @7.83.0_0
--->  Cleaning curl-ca-bundle
--->  Activating curl-ca-bundle @7.83.1_0
--->  Cleaning curl-ca-bundle
--->  Computing dependencies for curl
--->  Fetching archive for curl
--->  Attempting to fetch curl-7.83.1_0+http2+ssl.darwin_21.x86_64.tbz2 from https://packages.macports.org/curl
--->  Attempting to fetch curl-7.83.1_0+http2+ssl.darwin_21.x86_64.tbz2.rmd160 from https://packages.macports.org/curl
--->  Installing curl @7.83.1_0+http2+ssl
--->  Cleaning curl
--->  Computing dependencies for curl
--->  Deactivating curl @7.83.0_0+http2+ssl
--->  Cleaning curl
--->  Activating curl @7.83.1_0+http2+ssl
--->  Cleaning curl
--->  Computing dependencies for fheroes2
--->  Fetching archive for fheroes2
--->  Attempting to fetch fheroes2-0.9.15_0.darwin_21.x86_64.tbz2 from https://packages.macports.org/fheroes2
--->  Attempting to fetch fheroes2-0.9.15_0.darwin_21.x86_64.tbz2 from https://nue.de.packages.macports.org/fheroes2
--->  Attempting to fetch fheroes2-0.9.15_0.darwin_21.x86_64.tbz2 from https://fra.de.packages.macports.org/fheroes2
--->  Fetching distfiles for fheroes2
--->  Verifying checksums for fheroes2
--->  Extracting fheroes2
--->  Applying patches to fheroes2
--->  Configuring fheroes2
--->  Building fheroes2                                  
--->  Staging fheroes2 into destroot                     
--->  Installing fheroes2 @0.9.15_0
--->  Cleaning fheroes2
--->  Computing dependencies for fheroes2
--->  Deactivating fheroes2 @0.9.14_0
--->  Cleaning fheroes2
--->  Activating fheroes2 @0.9.15_0
--->  Cleaning fheroes2
--->  Updating database of binaries
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.
--->  Some of the ports you installed have notes:
  fheroes2 has the following notes:
    You will need to have a demo version or the full version of Heroes of Might and
    Magic II game to be able to play.
    
    If you do not have the original Heroes of Might and Magic II game, type the
    following command:
    
    fheroes2-install-demo
    
    This script will download and install all the necessary files from the demo
    version of the original Heroes of Might and Magic II game.
    
    If you have a legally purchased copy of the original game, type the following
    command:
    
    fheroes2-extract-resources
    
    This script will extract all the necessary resource files from the original
    Heroes of Might and Magic II game.
    
    As an alternative to the previous step, you can manually copy the subdirectories
    'ANIM', 'DATA', 'MAPS' and 'MUSIC' (some of them may be missing depending on the
    version of the original game) from the original game directory to the '.fheroes2'
    subdirectory of your home directory.
    
    When all prerequisites are in place, you can start the game by typing:
    
    fheroes2
    
    Please visit project website:
    
    https://ihhub.github.io/fheroes2/
    
    and read /opt/local/share/doc/fheroes2/README.txt for more information.
                                                                                                                                                                                                            
oleg@MacBook-Pro ~ % fheroes2
oleg@MacBook-Pro ~ % fheroes2
Free Heroes of Might and Magic II, version: 0.9.15
```

Could you please post here the output of the following command:

```bash
cat /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/games/fheroes2/Portfile
```

to make sure that you really have the latest version of this Portfile?

**Comment by JamonBull** [comment]
Sure thing! Thanks for taking the time.

---

# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4

PortSystem      1.0

PortGroup       github 1.0
PortGroup       cmake 1.1
PortGroup       compiler_blacklist_versions 1.0

github.setup    ihhub fheroes2 0.9.15

categories      games
platforms       darwin
maintainers     {@oleg-derevenetz gmail.com:oleg.derevenetz} openmaintainer
license         GPL-2

description     Free implementation of Heroes of Might and Magic II game engine
long_description \
                ${description}

homepage        https://ihhub.github.io/fheroes2/

checksums       rmd160  8f5ece18169f1704a49f16fba3f1a76487805abc \
                sha256  95c3a9b9afa7e7e77cdddc1d21d2dcdd0daf397b0a46aea75d6527dc560f7695 \
                size    3862349

depends_build-append \
                port:gettext

depends_lib     port:libpng \
                port:zlib \
                port:libsdl2 \
                port:libsdl2_mixer \
                port:libsdl2_image

patchfiles      patch-readme-txt.diff

compiler.cxx_standard \
                2011
compiler.thread_local_storage \
                yes
compiler.blacklist-append \
                {clang < 900}

post-destroot {
    copy ${destroot}${prefix}/share/doc/fheroes2/demo/download_demo_version.sh ${destroot}${prefix}/bin/fheroes2-install-demo
    copy ${destroot}${prefix}/share/doc/fheroes2/homm2/extract_homm2_resources.sh ${destroot}${prefix}/bin/fheroes2-extract-resources
}

notes "
You will need to have a demo version or the full version of Heroes of Might and
Magic II game to be able to play.

If you do not have the original Heroes of Might and Magic II game, type the
following command:

fheroes2-install-demo

This script will download and install all the necessary files from the demo
version of the original Heroes of Might and Magic II game.

If you have a legally purchased copy of the original game, type the following
command:

fheroes2-extract-resources

This script will extract all the necessary resource files from the original
Heroes of Might and Magic II game.

As an alternative to the previous step, you can manually copy the subdirectories
'ANIM', 'DATA', 'MAPS' and 'MUSIC' (some of them may be missing depending on the
version of the original game) from the original game directory to the '.fheroes2'
subdirectory of your home directory.

When all prerequisites are in place, you can start the game by typing:

fheroes2

Please visit project website:

${homepage}

and read ${prefix}/share/doc/fheroes2/README.txt for more information.
"

**Reply by JamonBull** [reply]
"The following ports are currently installed:
  fheroes2 @0.9.13_0
  fheroes2 @0.9.14_0
  fheroes2 @0.9.15_0 (active)"

But when I type "fheroes2" to launch the game, it launches 0.9.14. Strange!

**Reply by oleg-derevenetz** [reply]
@JamonBull

> But when I type "fheroes2" to launch the game, it launches 0.9.14. Strange!

It looks like you have it installed somewhere else from the another source. How about the output of the following command:

```bash
which fheroes2
```

?

**Comment by oleg-derevenetz** [comment]
@JamonBull Hmm, the Portfile looks good. What about the result of the following command:

```bash
port installed fheroes2
```

?

**Comment by JamonBull** [comment]
"/opt/homebrew/bin/fheroes2"

Aha! So how do I get rid of that?

**Reply by oleg-derevenetz** [reply]
@JamonBull it looks like you have both Homebrew and MacPorts on your system. Probably you should uninstall the version installed through Homebrew (something like `brew uninstall fheroes2`) and use just the version from MacPorts. In general, the idea of having several different package managers at the same time is not the best, it is better to choose only one package manager and stick to it.

**Comment by JamonBull** [comment]
This worked! Thank you so much. All sorted now. Running the latest version through MacPorts.
